sql/api/src/main/scala/org/apache/spark/sql/errors/ExecutionErrors.scala

```diff
@@ -120,13 +120,11 @@ private[sql] trait ExecutionErrors extends DataTypeErrorsBase {
       context: QueryContext = null): ArithmeticException = {
     val alternative = if (hint.nonEmpty) {
       s" Use '$hint' to tolerate overflow and return NULL instead."
-    } else ""
+    }
+    else s""" If necessary set ${SqlApiConf.ANSI_ENABLED_KEY} to "false" to bypass this error."""
```

Comment:
Let's don't embed error text to source code. We should allow to to edit `error-conditions.json` without repackaging Spark distro, and also we could support many other json files in the future (per language for instance).

Please, introduce sub-classes for hit and w/o it.

Reply:
Makes sense, until we figure out how to handle UnaryMinus and toIntExact sub-classes should work.